src/daemon/ca/child.rs

```diff
@@ -29,20 +45,35 @@ impl LastResponse {}
 /// and [ResourceClassName] are kept in the parent's [ResourceClass].
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct ChildDetails {
+    #[serde(default)]
+    state: ChildState,
     id_cert: IdCert,
     resources: ResourceSet,
-    used_keys: HashMap<KeyIdentifier, LastResponse>,
+    used_keys: HashMap<KeyIdentifier, UsedKeyState>,
 }
 
 impl ChildDetails {
     pub fn new(id_cert: IdCert, resources: ResourceSet) -> Self {
         ChildDetails {
+            state: ChildState::Active,
             id_cert,
             resources,
             used_keys: HashMap::new(),
         }
     }
 
+    pub fn is_suspended(&self) -> bool {
+        self.state == ChildState::Suspended
+    }
+
+    pub fn suspend(&mut self) {
+        self.state = ChildState::Suspended;
```

Comment:
This doesn't actually suspend the child but only mark it as suspended right? Would it be better then not to use the verb but to call this "mark_suspended" or such?